Frameworks/Foundation/NSHTTPURLResponse.mm

```diff
@@ -19,164 +19,116 @@
-    if (self = [super init]) {
-        if (headerFields != nil) {
-            CFDictionaryKeyCallBacks caseInsensitiveKeyChecker = kCFTypeDictionaryKeyCallBacks;
+/**
+ @Status Interoperable
+*/
+- (id)initWithURL:(NSURL*)url statusCode:(NSInteger)statusCode HTTPVersion:(NSString*)HTTPVersion headerFields:(NSDictionary*)headerFields {
+    int expectedContentLength = NSURLResponseUnknownLength;
+    NSString* mimeType = nil;
+    NSString* textEncodingName = nil;
+    NSString* filename = nil;
+    NSMutableDictionary* allHeaderFields = nil;
 
-            caseInsensitiveKeyChecker.equal = CFHTTPHeaderEqual;
-            caseInsensitiveKeyChecker.hash = CFHTTPHeaderHash;
+    if (headerFields != nil) {
+        CFDictionaryKeyCallBacks caseInsensitiveKeyChecker = kCFTypeDictionaryKeyCallBacks;
 
-            _allHeaderFields = (NSMutableDictionary*)
-                CFDictionaryCreateMutable(NULL, [headerFields count], &caseInsensitiveKeyChecker, &kCFTypeDictionaryValueCallBacks);
+        caseInsensitiveKeyChecker.equal = CFHTTPHeaderEqual;
+        caseInsensitiveKeyChecker.hash = CFHTTPHeaderHash;
 
-            //  Case insensitive dictionary
-            for (id key in [headerFields allKeys]) {
-                id val = [headerFields objectForKey:key];
-                [_allHeaderFields setObject:val forKey:key];
-            }
-        }
-        _url = url;
-        _statusCode = statusCode;
-        _expectedContentLength = expectedContentLength;
-    }
+        allHeaderFields = (NSMutableDictionary*)
+            CFDictionaryCreateMutable(NULL, [headerFields count], &caseInsensitiveKeyChecker, &kCFTypeDictionaryValueCallBacks);
 
-    return self;
-}
+        //  Case insensitive dictionary
+        for (id key in [headerFields allKeys]) {
+            id val = [headerFields objectForKey:key];
+            [allHeaderFields setObject:val forKey:key];
+        }
 
-/**
- @Status Interoperable
-*/
-- (id)initWithURL:(NSURL*)url statusCode:(NSInteger)statusCode HTTPVersion:(NSString*)HTTPVersion headerFields:(NSDictionary*)headerFields {
-    if (self = [self _initCommon:url statusCode:statusCode headerFields:headerFields expectedContentLength:-1]) {
-        _HTTPVersion = HTTPVersion;
-    }
-    return self;
-}
+        NSString* expectedLength = [allHeaderFields objectForKey:@"Content-Length"];
+        if (nil != expectedLength) {
+            expectedContentLength = [expectedLength intValue];
+        }
 
-/**
- @Status Interoperable
-*/
-- (id)initWithURL:(NSURL*)url statusCode:(NSInteger)statusCode headers:(id)headers expectedContentLength:(int)expectedContentLength {
-    self = [self _initCommon:url statusCode:statusCode headerFields:headers expectedContentLength:expectedContentLength];
-    return self;
-}
+        NSString* contentDisposition = [allHeaderFields objectForKey:@"Content-Disposition"];
+        NSRange filenameTagPosition = [contentDisposition rangeOfString:@"filename=" options:NSBackwardsSearch];
```

Comment:
+1 it will be good to capture all the header fields we care about and how we parse them

---
In reply to: [69051446](https://github.com/Microsoft/WinObjC/pull/595#discussion_r69051446) [](ancestors = 69051446)